firrtl/src/main/scala/firrtl/AttributeAnnotation.scala

```diff
@@ -0,0 +1,12 @@
+// SPDX-License-Identifier: Apache-2.0
+package firrtl
+
+import firrtl.annotations.{Named, SingleTargetAnnotation}
+
+/** Firrtl implementation for verilog attributes
+  * @param target       target component to tag with attribute
+  * @param description  Attribute string to add to target
+  */
+case class AttributeAnnotation(target: Named, description: String) extends SingleTargetAnnotation[Named] {
+  def duplicate(n: Named): AttributeAnnotation = this.copy(target = n, description = description)
```

Comment:
Q to @seldridge, do we still need to implement `duplicate`? this might be an old API from SFC world?

Reply:
You unfortunately, do as this is an abstract member. However, this method will never be used. We should remove this abstract member.

Reply:
There is still 1 case that makes this necessary. Because views (from DataView) may be a single data that targets multiple unrelated Data, we do a special renaming at the end of Chisel elaboration which uses this duplicate method. Cleaning this up requires changes to the annotation API--easiest if we ban user-defined annotations or otherwise we have to change the API for how the user creates annotations from Chisel (forcing them to contemplate views).

Reply:
> easiest if we ban user-defined annotations or otherwise we have to change the API for how the user creates annotations from Chisel (forcing them to contemplate views).

I think, in the future plan we should remove the user-defined annotations, and maybe use `binding` for views?